### PR TITLE
fix: alias model names collision with component types

### PR DESCRIPTION
**Pull request by Jshhhh**
*Issue #, if available:*

*Description of changes:*
- Alias model imports when colliding with imported components in same file

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: packages/codegen-ui-react/lib/helpers/index.ts [hunk 1]
```diff
@@ -14,3 +32,16 @@
   limitations under the License.
  */
 export const lowerCaseFirst = (input: string) => input.charAt(0).toLowerCase() + input.slice(1);
+
+export const addModelAlias = (componentTypes: string[], importCollection: ImportCollection, model: string) => {
```
Comment (awinberg-aws) [inline packages/codegen-ui-react/lib/helpers/index.ts:36]: nit: the rest of our code shouldn't need to be aware of the fact that the model name is potentially aliased. This's bit of collision detection/resolution is being abstracted away and all the rest of the app needs to know is that this is the way to add a model to the import collection

File: packages/codegen-ui-react/lib/helpers/index.ts [hunk 2]
```diff
@@ -14,3 +32,16 @@
...
+  if (componentTypes.includes(model) || isPrimitive(model)) {
+    let existingModelAlias = importCollection.getMappedAlias(ImportSource.LOCAL_MODELS, model);
+    if (!existingModelAlias) {
+      existingModelAlias = `${model}Aliased`;
```
Comment (awinberg-aws) [inline packages/codegen-ui-react/lib/helpers/index.ts:40]: prefer to suffix with `Model` as this is more in line with what a customer would write if found in this situation

File: packages/codegen-ui-react/lib/helpers/index.ts (hunk 1 above)
Comment (Jshhhh) [reply packages/codegen-ui-react/lib/helpers/index.ts:36]: In other words, the `importCollection` itself should be able to handle collisions and that the helper function is not needed?

File: packages/codegen-ui-react/lib/helpers/index.ts (hunk 2 above)
Comment (Jshhhh) [reply packages/codegen-ui-react/lib/helpers/index.ts:40]: I was leaning towards a suffix that is less likely to collide again

File: packages/codegen-ui-react/lib/helpers/index.ts (hunk 1 above)
Comment (awinberg-aws) [reply packages/codegen-ui-react/lib/helpers/index.ts:36]: I was originally commenting about the naming of the helper, but you took it to the next logical conclusion and I agree that a helper shouldn't be needed.

File: packages/codegen-ui-react/lib/helpers/index.ts (hunk 2 above)
Comment (awinberg-aws) [reply packages/codegen-ui-react/lib/helpers/index.ts:40]: How can you be certain that a model doesn't already collide when adding `Alias` to it?

Comment (Jshhhh) [reply packages/codegen-ui-react/lib/helpers/index.ts:40]: Yeah that is true as well. Ill add a short uuid like string to it

**Pull request closed without merging**